### PR TITLE
Skil 160

### PR DESCRIPTION
**Pull request by malloc-nbytes**
Adds a generic upload option to bulk upload students/TAs that deprecates studentImport. Added teamBulkUpload that uploads a series of teams, students, and TAs that deprecates teamImport. Has relating test files. Fixes error bugs on the front-end. Reworked some routes to use the new bulk upload functions.

### CODE REVIEW

File: BackEndFlask/Functions/__init__.py
Comment (Wrimo) [inline BackEndFlask/Functions/__init__.py:1]: What's the point of this empty file?

File: BackEndFlask/Functions/customExceptions.py
Comment (Wrimo) [inline BackEndFlask/Functions/customExceptions.py:1]: Do these need to be updated to the new Exception format that defines as seen in other files, or is this fine?

File: BackEndFlask/Functions/teamImport.py [hunk 1]
```diff
@@ -60,10 +60,11 @@ def teamcsvToDB(teamFile, owner_id, course_id):
 
                     if user is None:
                         return UserDoesNotExist
-                    
+
 
                     user_id = get_user_user_id_by_email(
                         ta_email)
+                    print(f"user_id: {user_id}")
```
Comment (Wrimo) [inline BackEndFlask/Functions/teamImport.py:67]: Leftover print

File: BackEndFlask/Functions/test_files/population_functions.py [hunk 2]
```diff
@@ -249,13 +319,14 @@ def deleteAllTeamsTeamMembers(course_id):
     
     for team in teams:
         team_id = team.team_id
-
-        team = delete_team(team_id)
         team_users = get_team_users_by_team_id(team_id)
+        print(f'DELETING: {team_users}')
```
Comment (Wrimo) [inline BackEndFlask/Functions/test_files/population_functions.py:323]: More leftover prints

File: BackEndFlask/Functions/test_files/test_teamBulkUpload.py [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+from population_functions import *
+import os
+
+def retrieveFilePath(fileName):
+    return os.getcwd() + os.path.join(os.path.sep, "Functions") + os.path.join(os.path.sep, "sample_files")+ os.path.join(os.path.sep, "teamBulkUpload-files") + os.path.join(os.path.sep, fileName)
+
+# test_wrong_extention_error
+#   - ensures that student_and_team_to_db
+#       - returns an error if given the incorrect file type
+def test_should_fail_with_wrong_extention_error(flask_app_mock):
```
Comment (Wrimo) [inline BackEndFlask/Functions/test_files/test_teamBulkUpload.py:12]: Is this function supposed to be mostly commented out? The assert true seems off.

File: BackEndFlask/controller/Routes/Bulk_upload_routes.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+            return create_bad_response("Unsuccessfully uploaded a .csv file! No file selected.", "users", 400)
+        extension = os.path.splitext(file.filename)
+        if(extension[1]!= ".csv" and extension[1] != ".xlsx"):
+            return create_bad_response(f"Unsuccessfully uploaded a {extension[1]} file! Wrong Format", "users", 400)
+
+        if request.args.get("course_id"):
+            course_id = int(request.args.get("course_id"))
+            user_id = int(request.args.get("user_id"))
+            directory = os.path.join(os.getcwd(), "Test")
+            os.makedirs(directory, exist_ok=True)
```
Comment (Wrimo) [inline BackEndFlask/controller/Routes/Bulk_upload_routes.py:27]: Personally, I would move this file handling stuff into a function called by the route, but it does not matter that much.

File: BackEndFlask/controller/Routes/Team_bulk_upload_routes.py [hunk 5]
```diff
@@ -27,35 +29,38 @@ def upload_team_csv():
...
             file_path = os.path.join(directory, file.filename)
             file.save(file_path)
-            result = teamImport.teamcsvToDB(file_path, 2, course_id)
 
-            if (result != "Upload successful!"):
-                shutil.rmtree(directory)
-                raise Exception("Failed to uploaded " + str(result))
+            user_id = int(request.args.get("user_id"))
+            teamBulkUpload.team_bulk_upload(file_path, user_id, course_id)
+
```
Comment (Wrimo) [inline BackEndFlask/controller/Routes/Team_bulk_upload_routes.py:35]: Do we need all this commented out code?

**Review by Wrimo — APPROVED** [review]
I made a number of comments, but they are all on fairly minor. I believe this could be pushed to master with minimal damage. Good work!

**Comment by malloc-nbytes** [comment]
@brianlugo1 I have addressed the issues that @Wrimo has brought up.

**Review by brianlugo1 — APPROVED** [review]
@sah0017 LGTM!

**Comment by brianlugo1** [comment]
@sah0017 This PR has received two approved code reviews! I am going to go ahead and merge this PR!

**Pull request merged**